tests/plugins/test_export_operator_manifests.py

```diff
@@ -0,0 +1,114 @@
+    return workflow
+
+
+def generate_archive(tmpdir):
+    archive_path = os.path.join(str(tmpdir), 'temp.tar')
+    archive_tar = tarfile.open(archive_path, 'w')
+    manifests_dir = os.path.join(str(tmpdir), 'manifests')
+    os.mkdir(manifests_dir)
+    open(os.path.join(manifests_dir, 'stub.json'), 'w').close()
+    archive_tar.add(manifests_dir, arcname='manifests')
+    archive_tar.close()
+    f = open(archive_path)
+    for line in f.read():
+        yield line
+    f.close()
+    os.unlink(archive_path)
+
+
+def mock_env(tmpdir, docker_tasker, has_label=True, label=True,
+             scratch=False, orchestrator=False, selected_platform=True):
+    build_json = {'metadata': {'labels': {'scratch': scratch}}}
+    flexmock(util).should_receive('get_build_json').and_return(build_json)
+    mock_dockerfile(tmpdir, has_label, label)
+    workflow = mock_workflow(tmpdir)
+    workflow.prebuild_results[PLUGIN_CHECK_AND_SET_PLATFORMS_KEY] = orchestrator
+    mock_stream = generate_archive(tmpdir)
+    plugin_conf = [{'name': ExportOperatorManifestsPlugin.key}]
+    if selected_platform:
+        plugin_conf[0]['args'] = {'allowed_platform': os.uname()[4], 'platform': os.uname()[4]}
+    runner = PostBuildPluginsRunner(docker_tasker, workflow, plugin_conf)
+
+    (flexmock(docker_tasker.d.wrapped)
+     .should_receive('create_container')
+     .with_args(workflow.image, command=["/bin/bash"])
+     .and_return({'Id': CONTAINER_ID}))
+
+    (flexmock(docker_tasker.d.wrapped)
+     .should_receive('remove_container')
+     .with_args(CONTAINER_ID))
+
+    (flexmock(docker_tasker.d.wrapped)
+     .should_receive('get_archive')
+     .with_args(CONTAINER_ID, '/manifests')
+     .and_return(mock_stream, {}))
+
+    return runner
+
+
+class TestExportOperatorManifests(object):
+    @pytest.mark.parametrize('scratch', [True, False])
+    @pytest.mark.parametrize('has_label', [True, False])
+    @pytest.mark.parametrize('label', [True, False])
+    @pytest.mark.parametrize('orchestrator', [True, False])
+    @pytest.mark.parametrize('selected_platform', [True, False])
+    def test_skip(self, docker_tasker, tmpdir, caplog, scratch, has_label,
+                  label, orchestrator, selected_platform):
+
+        runner = mock_env(tmpdir, docker_tasker, has_label=has_label, label=label, scratch=scratch,
+                          orchestrator=orchestrator, selected_platform=selected_platform)
+        result = runner.run()
+        if any([not has_label, not label, scratch, orchestrator, not selected_platform]):
+            assert 'Skipping' in caplog.text
+            assert result[PLUGIN_EXPORT_OPERATOR_MANIFESTS_KEY] is None
+
+    def test_export_archive(self, docker_tasker, tmpdir, caplog):
+        runner = mock_env(tmpdir, docker_tasker)
+        result = runner.run()
+        archive = result[PLUGIN_EXPORT_OPERATOR_MANIFESTS_KEY]
+        assert archive
+        assert archive.split('/')[-1] == 'operator_manifests.zip'
```

Comment:
Could we have a test-case that shows we get an error if there are no manifests to include in the archive? i.e. to test this: https://github.com/projectatomic/atomic-reactor/pull/1177#discussion_r260883597

Reply:
Done. Although, we had to mock the docker response.

Reply:
How about if the directory exists but is empty? Will that behave the some way?

Reply:
Added code to handle empty /manifests dir + tests